Modules/_winapi.c

```diff
@@ -774,12 +774,144 @@ gethandle(PyObject* obj, const char* name)
     return ret;
 }
 
+static PyObject* sortenvironmentkey(PyObject *module, PyObject *item) {
+    return _winapi_LCMapStringEx_impl(NULL, LOCALE_NAME_INVARIANT, LCMAP_UPPERCASE, item);
+}
+
+static PyMethodDef sortenvironmentkey_def = {
+    "sortenvironmentkey",
+    _PyCFunction_CAST(sortenvironmentkey),
+    METH_O,
+    ""
+};
+
+static PyObject *
+normalize_environment(PyObject* environment)
+{
+    PyObject *result = NULL, *keys = NULL, *keyfunc = NULL, *sort = NULL,
+             *args = NULL, *kwargs = NULL;
+
+    keys = PyMapping_Keys(environment);
+    if (keys == NULL) {
+        return NULL;
+    }
```

Comment:
```suggestion
    }
    asssert(PyList_CheckExact(keys));
```
The `PyMapping_Keys` returns an exact list. This assertion is to make sure this is ineed the case.

Reply:
You mean remove the `if (keys == NULL) { ... }` and changed to using the `asssert(PyList_CheckExact(keys))`?

Reply:
If `PyMapping_Keys`'s [document](https://docs.python.org/3.9/c-api/mapping.html#c.PyMapping_Keys) guarantees it will returns a list, I think this assert is not necessary?